wallet/wallet-cli/src/main.rs

```diff
@@ -21,28 +21,31 @@ mod commands;
 mod config;
 mod console;
 mod errors;
-mod helpers;
 mod repl;
-mod wallet_init;
 
-use std::{net::SocketAddr, str::FromStr, sync::Arc};
+use std::{io::BufRead, net::SocketAddr, str::FromStr, sync::Arc};
 
 use clap::Parser;
 use common::chain::config::ChainType;
 use config::WalletCliArgs;
 use console::ConsoleContext;
-use dialoguer::theme::ColorfulTheme;
+use crossterm::tty::IsTty;
 use errors::WalletCliError;
 use tokio::sync::mpsc;
-use utils::default_data_dir::{default_data_dir_for_chain, prepare_data_dir};
-use wallet::Wallet;
+use utils::default_data_dir::default_data_dir_for_chain;
 use wallet_controller::cookie::load_cookie;
 
 const COOKIE_FILENAME: &str = ".cookie";
 
-async fn run(output: &ConsoleContext) -> Result<(), WalletCliError> {
-    logging::init_logging::<&std::path::Path>(None);
+enum Mode {
+    Interactive {
+        printer: reedline::ExternalPrinter<String>,
+    },
+    NonInteractive,
+    Commands(std::fs::File),
```

Comment:
`CommandsList(Vec<String>)`.

This makes it possible to run unit tests with this. Then the only difference is that you do an std::fs::read_lines() or similar and input here.

The idea here is we limit having interaction with files directly as much as possible. Dealing with the file system invites all kinds of problems.

Reply:
Updated. I still need to do IO to read from `stdin`, but this is now hidden behind generic `Iterator`:
```
pub fn run(
    lines: impl Iterator<Item = String>,
    output: &ConsoleContext,
    event_tx: mpsc::UnboundedSender<Event>,
    exit_on_error: bool,
) -> Result<(), WalletCliError> {
```

Reply:
Cool. No problem there (I'll look at that part indepndently).

Though please call the arm "CommandsList", not "Commands".